library/packages/src/modules/PackageCallbacks.rb

```diff
@@ -2402,6 +2467,7 @@ def ProgressProgress(id, val_raw, val_percent)
     # @param [Array<String>] stages Descriptions of the stages
     # @param [String] help Help text describing the process
     def ProcessStart(task, stages, help)
+      log.info("Starting callback #{self.class}::#{__method__}") if ENV["Y2DEBUG_CALLBACKS"] == "1"
```

Comment:
well, it looks like a lot of code and lot of duplication. What about adding hook when method is called?
see https://stackoverflow.com/questions/3236194/defining-method-called-how-do-i-make-a-hook-method-which-gets-called-every-t